### ISSUE TITLE
[Bug] Interacting with Hex Editor or opening Pattern Editor disables keyboard navigation

### ISSUE THREAD

**Issue by bbb651** [WerWolv/ImHex#2763]
### Operating System

Linux

### What's the issue you encountered?

Keyboard navigation gets globally disabled when Interacting with Hex Editor or opening Pattern Editor, so pretty much always.

It's pretty straight forward why this happens, [keyboard navigation is turned on when initializing imgui
](https://github.com/search?q=repo%3AWerWolv%2FImHex%20%22ImGuiConfigFlags_NavEnableKeyboard%22&type=code), then [calls to `discardNavigationRequests()`](https://github.com/search?q=repo%3AWerWolv%2FImHex%20discardNavigationRequests&type=code) disable it and it's never enabled again.

### How can the issue be reproduced?

- Interact with Hex Editor or open Pattern Editor
- Keyboard navigation doesn't work in e.g. Find search results and Pattern Data

### ImHex Version

1.38.1

### ImHex Build Type

- [ ] Nightly or built from sources

### Installation type

ArchLinux AUR `imhex-bin` package

**Comment by paxcut** [comment]
The title of this issue is somewhat misleading as it seems to imply that the keyboard may not be used in ImHex to perform navigation commands like page up/down arrow keys, etc... The issue also incorrectly claims that keyboard navigation is disabled globally when using the hex editor or the pattern editor. The code pointing to where the disabling takes place is in reality only disabling it for the hex editor and the pattern editor windows. They do that because they both implement their own navigation commands using shortcuts that can be customized in preferences.

ImGui only provides very basic keyboard navigation and it is restricted to work only from within widgets. Thus you can use the arrow keys to navigate inside text input widget fields, but you can't activate buttons or navigate menus using the keyboard by default.Those interactions must be explicitly implemented. ImGui does not provide for means to disable keyboard navigation globally and flags are restricted to work only within windows,  widgets or whatever the flag context is.

**Comment by bbb651** [comment]
> The title of this issue is somewhat misleading as it seems to imply that the keyboard may not be used in ImHex to perform navigation commands like page up/down arrow keys, etc...

I was specifically talking about the keyboard navigation builtin to imgui for things like lists and tree, hence the Find search results and Pattern Data examples.

> The code pointing to where the disabling takes place is in reality only disabling it for the hex editor and the pattern editor windows.

This is not true, this seems to be the intent of the code but it's not what it does. It's very easy to reproduce and the documentation makes it very clear it's global to the application and not the window:
```c
struct ImGuiIO;                     // Main configuration and I/O between your application and ImGui (also see: ImGuiPlatformIO)
```
